src/Entity/FieldableEdgeEntityBaseTrait.php

```diff
@@ -377,13 +377,25 @@ public function set($field_name, $value, $notify = TRUE) {
     // (We do not change the structure of values that does not belong to
     // base fields).
     if (is_array($value) && $this->getFieldDefinition($field_name) instanceof BaseFieldDefinition && $this->getFieldDefinition($field_name)->getCardinality() === 1) {
-      $exists = FALSE;
-      $value = NestedArray::getValue($value, ['0', 'value'], $exists);
-      if (!$exists) {
-        $value = NestedArray::getValue($value, ['value'], $exists);
-        if (!$exists) {
-          // We should know about this.
-          throw new EdgeFieldException(sprintf('Unable to retrieve value of %s base field on %s.', $field_name, get_called_class()));
+      if (count($value) === 0) {
+        $value = '';
+      }
+      else {
+        $exists = FALSE;
+        $new_value = NestedArray::getValue($value, ['0', 'value'], $exists);
+        if ($exists) {
+          $value = $new_value;
+        }
+        else {
+          $new_value = NestedArray::getValue($value, ['value'], $exists);
+          if ($exists) {
+            $value = $new_value;
+          }
+          else {
+            $called_class = get_called_class();
+            $exported_value = var_export($value, TRUE);
```

Comment:
Could it be a security risk that the exception (which probably gets logged) contains a value of the field?

Reply:
It exposes the field's content, that's true. I am not sure if it can be a security problem, since you need admin permissions to access the logs.

If this part of the code runs then it is always a bug (the content of the field won't be saved), and it this makes troubleshooting so much easier.